### PR TITLE
feat(build-release-android): update the API consumption for accessibility insights android service

### PR DESCRIPTION
**Pull request by waabid**
#### Description of changes

This makes 2 change:
1. It updates to use "AccessibilityInsights" in it's API usage. This is the first commit.
2. Update the connect experience to utilize our config API. This is the second commit.

I don't mind breaking these into two PRs if we prefer.

Also note that the changes required me to make modifications such that the mock android task does not fail (necessitates change in folder structure).

#### Pull request checklist

- [n/a] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). Check workflow guide at: `<rootDir>/docs/workflow.md`
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/tests/miscellaneous/mock-axe-android/AccessibilityInsights/config.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (dbjorge) [inline src/tests/miscellaneous/mock-axe-android/AccessibilityInsights/config.json]: Should the folder name and `package.json` script names still referencing "axe-android" also be updated

File: src/tests/miscellaneous/mock-axe-android/README.md [hunk 2]
```diff
@@ -7,5 +7,5 @@ Licensed under the MIT License.
 
 This directory is for the benefit of the "yarn serve:mock-axe-android" command (see package.json).
 
-The specific folder/file names here ("/axe/result.json") are important to leave as-is because they
-correspond to the endpoint that Axe for Android normally creates ("http://localhost:port/axe/result").
+The specific folder/file names here ("/AccessibilityInsights/result.json") are important to leave as-is because they
+correspond to the endpoint that Axe for Android normally creates ("http://localhost:port/AccessibilityInsights/result").
```
Comment (DaveTryon) [inline src/tests/miscellaneous/mock-axe-android/README.md]: Seems like this ought to be renamed?
```suggestion
correspond to the endpoint that Axe for Android normally creates ("http://localhost:port/AccessibilityInsights/result").
```

Comment (DaveTryon) [inline src/tests/miscellaneous/mock-axe-android/README.md]: "Axe for Android" should probably be updated

File: docs/workflow.md [hunk 3]
```diff
@@ -26,12 +26,12 @@ yarn start:unified:dev
 
 To debug the built app in vscode, you can run the `Debug electron main process with --remote-debugging-port=9222` configuration. Once this is running, you can debug the renderer process via the `Attach debugger to electron renderer process` configuration.
 
-Most of the functionality of the Electron app relies on connecting to a running [Axe for Android](https://www.deque.com/axe/axe-for-android/) service.
+Most of the functionality of the Electron app relies on connecting to Accessibility Insights for Android Service.
```
Comment (DaveTryon) [inline docs/workflow.md]: Reword this sentence to flow better?
```suggestion
Most of the functionality of the Electron app relies on connecting to a device running the Accessibility Insights for Android Service.
```

File: docs/workflow.md [hunk 4]
```diff
@@ -26,12 +26,12 @@ yarn start:unified:dev
...
 
-To simulate Axe for Android (running on port 9051) for local testing of the app, use:
+To simulate Accessibility Insights for Android Service (running on port 9051) for local testing of the app, use:
 
 ```sh
```
Comment (DaveTryon) [inline docs/workflow.md]: Reword?
```suggestion
To simulate the Accessibility Insights for Android Service (running on port 9051) for local testing of the app, use:

```sh
```

File: src/tests/miscellaneous/mock-service-for-android/README.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+-->
+
+# mock-service-for-android test content
+
+This directory is for the benefit of the "yarn serve:mock-service-for-android" command (see package.json).
+
+The specific folder/file names here ("/AccessibilityInsights/result.json") are important to leave as-is because they
+correspond to the endpoint that Accessibility Insights for Android Service normally creates ("http://localhost:port/AccessibilityInsights/result").
```
Comment (DaveTryon) [inline src/tests/miscellaneous/mock-service-for-android/README.md]: slight reword?
```suggestion
correspond to the endpoints that the Accessibility Insights for Android Service normally creates ("http://localhost:port/AccessibilityInsights/result").
```

File: src/electron/platform/android/device-config-fetcher.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+export class DeviceConfig {
+    constructor(readonly rawData: any) {}
+
+    public get deviceName(): string {
+        try {
+            return this.rawData.deviceName;
+        } catch {
+            return null;
+        }
```
Comment (devaradhanm) [inline src/electron/platform/android/device-config-fetcher.ts]: can't you just do this.rawData?.deviceName?

File: src/electron/platform/android/device-config-fetcher.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    public get appIdentifier(): string {
+        try {
+            return this.rawData.packageName;
```
Comment (devaradhanm) [inline src/electron/platform/android/device-config-fetcher.ts]: same as above

**Pull request merged**